packages/core/capabilities/core-capabilities-server-internal/src/resolve_capabilities.ts

```diff
@@ -82,3 +130,39 @@ function recursiveApplyChanges<
       return acc;
     }, {} as TDestination);
 }
+
+const getSwitchersToUseForPath = (path: string, switchers: SwitcherWithId[]): string[] => {
+  const switcherIds: string[] = [];
+  switchers.forEach((switcher) => {
+    if (switcher.capabilityPath.some((switcherPath) => pathsIntersect(path, switcherPath))) {
+      switcherIds.push(switcher.id);
+    }
+  });
+  return switcherIds;
+};
+
+const buildGetAggregatedSwitchers =
+  (
+    getSwitcherForPath: ForPathSwitcherResolver,
+    switcherMap: Map<string, SwitcherWithId>
+  ): AggregatedSwitchersResolver =>
+  (capabilityPaths: string[]): CapabilitiesSwitcher[] => {
+    // find switchers that should be applied for the provided capabilityPaths
+    const allSwitchers = [...switcherMap.values()];
+    const switcherIdsToApply = new Set<string>();
+    capabilityPaths.forEach((path) => {
+      getSwitcherForPath(path, allSwitchers).forEach((switcherId) =>
+        switcherIdsToApply.add(switcherId)
+      );
+    });
+    const switchersToApply = [...switcherIdsToApply].reduce((list, switcherId) => {
+      list.push(switcherMap.get(switcherId)!);
+      return list;
+    }, [] as SwitcherWithId[]);
```

Comment:
nitnit, maybe a little simpler? or is `.reduce` + `.push` more performant?

```ts
[...switcherIdsToApply].map((switcherId) => {
      return switcherMap.get(switcherId)!;
    }
```

Reply:
Nope, you're right, I just missed the obvious `map` call I could use here 😄